node/src/client/mod.rs

```diff
@@ -181,6 +201,72 @@ impl<N: Network, C: ConsensusStorage<N>> Client<N, C> {
         }));
     }
 
+    /// Initializes transaction verification.
+    fn initialize_transaction_verification(&self) {
+        // Start the transaction verification loop.
+        let node = self.clone();
+        self.handles.lock().push(tokio::spawn(async move {
+            loop {
+                // If the Ctrl-C handler registered the signal, stop the node.
+                if node.shutdown.load(Relaxed) {
+                    info!("Shutting down transaction verification");
+                    break;
+                }
+
+                // Sleep briefly to allow transactions to be validated.
+                tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+
+                // Determine the length of the queue.
+                let queue_len = node.transaction_queue.lock().len();
+
+                // If we don't have any transactions to verify, try again later.
+                if queue_len == 0 {
+                    continue;
+                }
+
+                // Determine how many transactions we want to verify.
+                let num_transactions = VERIFICATION_CONCURRENCY_LIMIT.min(queue_len);
+
+                // Check if we have room to verify the transactions and update the counter accordingly.
+                let previous_verification_counter = node.verification_counter.fetch_update(Relaxed, Relaxed, |c| {
+                    // If we are already verifying sufficient transactions, don't verify any more for now.
+                    if c >= VERIFICATION_CONCURRENCY_LIMIT {
+                        None
+                    // If we have space to verify more transactions, verify as many as we can.
+                    } else {
+                        // Consider verifying the full transaction queue, but limit to the concurrency limit.
+                        Some((c + queue_len).min(VERIFICATION_CONCURRENCY_LIMIT))
+                    }
+                });
+
+                // Determine how many transactions we cán verify.
+                let num_transactions = match previous_verification_counter {
+                    // Determine how many transactions we can verify.
+                    Ok(previous_value) => num_transactions.saturating_sub(previous_value),
+                    // If we are already verifying sufficient transactions, don't verify any more for now.
+                    Err(_) => continue,
+                };
+
+                // For each transaction, spawn a task to verify it.
+                for _ in 0..num_transactions {
+                    if let Some((_, (peer_ip, serialized, transaction))) = node.transaction_queue.lock().pop_lru() {
+                        let _node = node.clone();
+                        tokio::spawn(async move {
```

Comment:
I wonder if these task handles shouldn't be accounted for in case of a shutdown :thinking:

Reply:
Hmm, interesting idea, but we don't currently pass a handler into the transaction verifier, so adding an extra shutdown check would not help much I think without a large refactor. We also don't do this for the [solution verifier](https://github.com/AleoHQ/snarkOS/blob/max_concurrent_deploy_verification/node/src/client/router.rs#L275).

Reply:
as long as we are still able to shut down normally (i.e. any ongoing tx verification isn't slowing things down too much), this shouldn't be an issue :ok_hand: